### DISCUSSION TITLE
Component-based tweens

### DISCUSSION THREAD

**Discussion by f1yingbanana** [jeffreylanters/unity-tweens#24] [Ideas]
I'm wondering if it'd be possible to write a generic component that can be added to any game object so no code is required for an animation. This would be very convenient in cases where some UI elements need to be animated, but don't have dedicated behaviour scripts to put the code in.

E.g. having a list of tweens (or maybe one per component, but allowing multiple components?) that are editable from the inspector, taking in a target (or maybe implicitly, a component on the same game object), the type of tween (from the list of available tweens), then the fields/curves/values?

Also, this looks like a very clean package to use, thanks for sharing this!

**Comment by jeffreylanters** [comment]
Hi! Thanks for your suggestion. I've thought about it a lot but hadn't implemented it yet because in the idea of the package is to be a script based tweening engine. But if there is an audience for a component solution, I'm willing to implement this. I'm going to look into this a little further and might implement it out of the box, or via a separate package which would add a component layer. I'll keep you posted on this thread.
